types/src/vm/configs.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::on_chain_config::{
+    randomness_api_v0_config::{AllowCustomMaxGasFlag, RequiredGasDeposit},
+    ConfigStorage, FeatureFlag, Features, OnChainConfig, TimedFeatureFlag, TimedFeatures,
+};
+use move_binary_format::deserializer::DeserializerConfig;
+use move_bytecode_verifier::VerifierConfig;
+use move_vm_runtime::config::VMConfig;
+
+pub fn aptos_prod_deserializer_config(
+    features: &Features,
+    gas_feature_version: u64,
+) -> DeserializerConfig {
+    // Note: binary format v6 adds a few new integer types and their corresponding instructions.
+    //       Therefore, it depends on a new version of the gas schedule and cannot be allowed if
+    //       the gas schedule hasn't been updated yet.
+    let max_binary_format_version =
+        features.get_max_binary_format_version(Some(gas_feature_version));
+
+    DeserializerConfig::new(
+        max_binary_format_version,
+        features.get_max_identifier_size(),
+    )
+}
+
+pub fn aptos_prod_verifier_config(features: &Features) -> VerifierConfig {
+    let use_signature_checker_v2 = features.is_enabled(FeatureFlag::SIGNATURE_CHECKER_V2);
+    let sig_checker_v2_fix_script_ty_param_count =
+        features.is_enabled(FeatureFlag::SIGNATURE_CHECKER_V2_SCRIPT_FIX);
+
+    VerifierConfig {
+        max_loop_depth: Some(5),
```

Comment:
I'm not sure if that makes sense here to extract some constants out as actual constants.